pkg/component/etcd/lease/lease.go

```diff
@@ -75,10 +82,23 @@ func (c *component) Destroy(ctx context.Context) error {
 	return nil
 }
 
+func (c *component) GetPeerURLTLSEnabledStatus(ctx context.Context) (bool, error) {
+	tlsEnabledValues, err := c.getTLSEnabledAnnotationValues(ctx, c.values.EtcdName)
+	if err != nil {
+		return false, err
+	}
+	tlsEnabled := true
+	for _, v := range tlsEnabledValues {
+		tlsEnabled = tlsEnabled && v
+	}
```

Comment:
AFAIU, you are checking the annotation in lease object of all etcd clusters  member , right but why ? 
Is there a reason behind it ?

Reply:
The idea is to check if there is any member who still has a value of false. If the annotation is non-existent then it is ignored because its possible that the etcdbr has still not been able to set this annotation.
In the case the annotation is false then it requires 2 restarts in total.

Having said that, ideally this will return false when there is only a single-member etcd in case of gardener. Because today a single member etcd cluster is launched with a peer URL which does not have TLS enabled. When there is a multi-node cluster then it will be ensured from gardener that all new members will always have peer URL TLS enabled.

However, the above assumption is only relevant in case of gardener.

Reply:
ok, got it